iceoryx_posh/test/moduletests/test_popo_server_port_roudi.cpp

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "test_popo_server_port_common.hpp"
+
+namespace iox_test_popo_server_port
+{
+constexpr iox::units::Duration ServerPort_test::DEADLOCK_TIMEOUT;
+
+TEST_F(ServerPort_test, GetClientTooSlowPolicyReturnsCorrectValues)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7090916c-57c5-4ef4-9876-87e58ab64058");
+
+    auto& sutWithDiscardOldestData = serverPortWithOfferOnCreate;
+    auto& sutWithWaitForConsumer = serverOptionsWithWaitForConsumerClientTooSlowPolicy;
+
+    EXPECT_THAT(sutWithDiscardOldestData.portRouDi.getClientTooSlowPolicy(),
+                Eq(ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA));
+    EXPECT_THAT(sutWithWaitForConsumer.portRouDi.getClientTooSlowPolicy(),
+                Eq(ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER));
+}
+
+TEST_F(ServerPort_test, ReleaseAllChunksWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f29f8890-c106-470d-820f-32eeea694f17");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    // produce chunks for the chunk receiver
+    pushRequests(sut.requestQueuePusher, QUEUE_CAPACITY);
+
+    // produce chunks for the chunk sender
+    allocateResponseWithRequestHeaderAndThen(sut, [&](const auto, auto res) { sut.portUser.sendResponse(res); });
+
+    EXPECT_THAT(getNumberOfUsedChunks(), Ne(0U));
+
+    sut.portRouDi.releaseAllChunks();
+
+    // this is not part of the server port but holds the chunk from `sendResponse`
+    clientResponseQueue.clear();
+
+    EXPECT_THAT(getNumberOfUsedChunks(), Eq(0U));
+}
+
+// BEGIN tryGetCaProMessage tests
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnOfferWhenNotOfferingHasCaProMessageTypeOffer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8944621b-4753-413b-bee0-a714fa4324c8");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    sut.portUser.offer();
+
+    sut.portRouDi.tryGetCaProMessage()
+        .and_then([&](const auto& caproMessage) { EXPECT_THAT(caproMessage.m_type, Eq(CaproMessageType::OFFER)); })
+        .or_else([&]() { FAIL() << "Expected CaPro message but got none"; });
+}
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnOfferWhenOfferingHasNullopt)
```

Comment:
The name is a bit misleading. What about `TryGetCaProMessageOnOfferWhenPortAlreadyOffers`